src/main/java/dota/buff/service/impl/MatchServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package dota.buff.service.impl;
+
+import com.github.wannesvr.core.Dota2ApiClient;
+import com.github.wannesvr.core.model.match.*;
+import com.github.wannesvr.core.request.match.MatchDetailRequest;
+
+import dota.buff.model.HeroDTO;
+import dota.buff.model.MatchDTO;
+import dota.buff.model.PlayerDTO;
+import dota.buff.service.HeroService;
+import dota.buff.service.MatchService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MatchServiceImpl implements MatchService {
+
+    private static final String API_KEY = "B946D0A6B0726327F1C9D44095CD11C0";
+    private static final Dota2ApiClient client = new Dota2ApiClient(API_KEY);
+    private static final HeroService heroService = new HeroServiceImpl();
+
+    @Override
+    public MatchDTO getMatchById(long matchId) {
+        MatchDetail matchDetail = client.send(getMatchDetailRequest(matchId));
+        return new MatchDTO(
+                matchDetail.getMatchId(),
+                matchDetail.getGameMode().toString(),
+                convertAllMatchPlayers(matchDetail),
+                convertAllMatchHeroes(convertAllMatchPlayers(matchDetail)),
+                matchDetail.isRadiantWin(),
+                matchDetail.getDuration());
```

Comment:
I think would be better to transfer matchDetail into MatchDTO constructor and incapsulate chosen params there